src/rust/engine/src/externs/interface.rs

```diff
@@ -1626,17 +1626,69 @@ fn single_file_digests_to_bytes<'py>(
   })
 }
 
-fn ensure_path_doesnt_exist(path: &Path) -> io::Result<()> {
-  match std::fs::remove_file(path) {
-    Ok(()) => Ok(()),
+/// Delete `path` and anything under it (if a directory), without following symlinks, returning all
+/// paths removed in `deleted` (even if there was an error part way through)
+fn ensure_path_doesnt_exist(deleted: &mut Vec<PathBuf>, path: PathBuf) -> io::Result<()> {
+  match std::fs::remove_file(&path) {
+    Ok(()) => {
+      deleted.push(path.to_owned());
+      Ok(())
+    }
     Err(e) if e.kind() == io::ErrorKind::NotFound => Ok(()),
     // Always fall through to remove_dir_all unless the path definitely doesn't exist, because
     // std::io::ErrorKind::IsADirectory is unstable https://github.com/rust-lang/rust/issues/86442
     //
     // NB. we don't need to check this returning NotFound because remove_file will identify that
     // above (except if there's a concurrent removal, which is out of scope)
-    Err(_) => std::fs::remove_dir_all(path),
+    Err(_) => remove_dir_all_recursive(deleted, path),
+  }
+}
+
+// This is very similar to std::fs::remove_dir_all's implementation, but we need to extract the
+// paths that were deleted.
```

Comment:
Oops. Yea, as mentioned: this isn't necessary =x

Reply:
Ah, okay, I think I misinterpreted your comment, all good!